core/services/RestService.js

```diff
@@ -15,27 +15,13 @@ module.exports = {
       if (!ctx.params.id) {
         ctx.throw(400, "id is required")
       }
-      let otmPopulate = []
-      let mtmPopulate = []
+      let populateArgs = []
```

Comment:
Look, now your parsing logic is in two different modules. First part is here, and second part is inside the Adapter. Why? Move ALL of the parsing logic into the Adapter and just pass the raw populate parameter from here.